packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -174,6 +174,125 @@ class Plugin {
     );
   }
 
+  /// Queries the `pubpsec` and returns the `platforms` sub map.
+  ///
+  /// Throws ToolExit if cannot find the `platforms:` map.
+  static YamlMap getPlatformsYamlMap(YamlMap pubspec) {
+    if (pubspec == null) {
+       throwToolExit('Invalid pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap flutterConfig = pubspec['flutter'] as YamlMap;
+    if (flutterConfig == null) {
+      throwToolExit('Did not find valid `flutter` map in pubspec.yaml', exitCode: 2);
+    }
+    final YamlMap pluginConfig = flutterConfig['plugin'] as YamlMap;
+    if (pluginConfig == null) {
+      throwToolExit('Did not find valid `plugin` map in pubspec.yaml', exitCode: 2);
+    }
+    if (pluginConfig['platforms'] == null) {
+      throwToolExit('Did not find valid `platforms` map in pubspec.yaml', exitCode: 2);
+    }
+
+    return pluginConfig['platforms'] as YamlMap;
```

Comment:
missing a type check here, platforms could be anything

Reply:
done